src/input-files.cc

```diff
@@ -666,7 +666,7 @@ void ObjectFile<E>::sort_relocations(Context<E> &ctx) {
         continue;
 
       std::span<ElfRel<E>> rels = isec->get_rels(ctx);
-      if (!std::is_sorted(rels.begin(), rels.end(), less))
+      if (!std::ranges::is_sorted(rels, less))
```

Comment:
I avoided std::ranges because it tends to not work with older compilers or the standard C++ libraries.

Reply:
I'm happy to drop these commits, but just out of curiosity I'll go double-check with various Docker images whether this is still an issue.

My first testing candidate, Ubuntu 20.04, fails for an unrelated reason:
```
CMake Error at third-party/mimalloc/CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.18 or higher is required.  You are running version 3.16.3
```

Reply:
Since the bundled mimalloc library requires CMake 3.18, building on **Ubuntu 20.04** has been broken since commit d2870ac57428be6f5f342b1f75b63d50cc1c9828 in July.

Building on **Debian 11** (which ships CMake 3.18) works fine with GCC 10, even with the range algorithms.
With Clang 11 the build fails for various reasons, among them the range algorithms (my fault) and the use of concepts (not my fault).

Building on **Ubuntu 22.04** with Clang 12, 13 and 14 fails for the same reasons.

Building on **Debian 12** with Clang 13 and 14 fails for the same reasons.
With Clang 15 it fails only due to the range algorithms (my fault).
With Clang 16 everything works fine.

Reply:
Bottom line:
* The minimum required versions are currently **CMake 3.18** and either **GCC 10** or **Clang 15**.
* Definitely no range algorithms for now; they would raise the minimum required Clang version to 16.